micropip/_compat_in_pyodide.py

```diff
@@ -21,6 +22,27 @@
         raise
     # Otherwise, this is pytest test collection so let it go.
 
+if IN_BROWSER:
+
+    async def _pyfetch(urls: str, **kwargs):
+        if "signal" in kwargs:
+            return await pyfetch(urls, **kwargs)
+
+        controller = AbortController.new()
+        kwargs["signal"] = controller.signal
+
+        async def fetch_with_abort():
+            try:
+                return await pyfetch(urls, **kwargs)
+            except CancelledError:
+                controller.abort()
+                raise
+
+        return await fetch_with_abort()
```

Comment:
By the way, I think maybe this logic can be written into `pyfetch` itself, because `cancel` is more pythonic than using a `AbortController`, and as I understand it, `pyfetch` is a pythonic wrapper for `js.fetch`

Reply:
PR welcome!